ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandStoreTest.java

```diff
@@ -264,72 +270,74 @@ public void shouldFilterNullPriorCommand() {
 
-    final ProducerRecord<CommandId, Command> record = recordCapture.getValue();
+    // Then:
+    final ProducerRecord<CommandId, Command> record = recordCaptor.getValue();
     assertThat(record.key(), equalTo(commandId));
     assertThat(record.value().getStatement(), equalTo(statementText));
-    assertThat(record.value().getOverwriteProperties(), equalTo(overrideProperties));
-    assertThat(record.value().getOriginalProperties(), equalTo(ksqlConfig.getAllConfigPropsWithSecretsObfuscated()));
+    assertThat(record.value().getOverwriteProperties(), equalTo(OVERRIDE_PROPERTIES));
+    assertThat(record.value().getOriginalProperties(),
+        equalTo(KSQL_CONFIG.getAllConfigPropsWithSecretsObfuscated()));
   }
 
-  private void setupConsumerToReturnCommand(final CommandId commandId, final Command command) {
-    reset(commandConsumer);
-    expect(commandConsumer.poll(anyObject(Duration.class))).andReturn(
-        buildRecords(commandId, command)
-    ).times(1);
-    replay(commandConsumer);
+  @Test
+  public void shouldIncludeCommandSequenceNumberInSuccessfulQueuedCommandStatus() {
+    // When:
+    final QueuedCommandStatus commandStatus =
+        commandStore.enqueueCommand(statementText, statement, KSQL_CONFIG, OVERRIDE_PROPERTIES);
+
+    // Then:
+    assertThat(commandStatus.getCommandSequenceNumber(), equalTo(recordMetadata.offset()));
   }
 
-  private CommandStore createCommandStoreThatAssignsSameId(final CommandId commandId) {
-    final CommandIdAssigner commandIdAssigner = mock(CommandIdAssigner.class);
-    expect(commandIdAssigner.getCommandId(anyObject())).andStubAnswer(
-        () -> new CommandId(commandId.getType(), commandId.getEntity(), commandId.getAction())
-    );
-    replay(commandIdAssigner);
-    return createCommandStore(commandIdAssigner);
+  @Test
+  public void shouldWaitOnSequenceNumberFuture() throws Exception {
+    // When:
+    commandStore.ensureConsumedUpThrough(2, TIMEOUT);
+
+    // Then:
+    verify(future).get(eq(TIMEOUT), eq(TimeUnit.MILLISECONDS));
   }
 
-  private CommandStore createCommandStore() {
-    return createCommandStore(new CommandIdAssigner(new MetaStoreImpl(new InternalFunctionRegistry())));
+  @Test
+  public void shouldThrowExceptionOnTimeout() throws Exception {
+    // Given:
+    expectedException.expect(TimeoutException.class);
+    expectedException.expectMessage(String.format(
+        "Timeout reached while waiting for command sequence number of 2. (Timeout: %d ms)", TIMEOUT));
+
+    when(future.get(anyLong(), any(TimeUnit.class))).thenThrow(new TimeoutException());
+
+    // When:
+    commandStore.ensureConsumedUpThrough(2, TIMEOUT);
   }
 
-  private CommandStore createCommandStore(final CommandIdAssigner commandIdAssigner) {
-    return new CommandStore(
-        COMMAND_TOPIC,
-        commandConsumer,
-        commandProducer,
-        commandIdAssigner);
+  @Test
+  public void shouldCompleteFuturesWhenGettingNewCommands() {
```

Comment:
It's important it does the `completeFuturesUpThroughSequenceNumber` _before_ the `poll`, right? And its important what position is passed, so...

```java
  @Test
  public void shouldCompleteFuturesWhenGettingNewCommands() {
    // Given:
    when(commandConsumer.position(COMMAND_PARTITION_0)).thenReturn(22L);

    // When:
    commandStore.getNewCommands();

    // Then:
    final InOrder inOrder = inOrder(sequenceNumberFutureStore, commandConsumer);
    inOrder.verify(sequenceNumberFutureStore).completeFuturesUpThroughSequenceNumber(eq(21L));
    inOrder.verify(commandConsumer).poll(any());
  }
```